frontend/src/App.tsx

```diff
@@ -1,26 +1,1041 @@
+                "strCreativeCommonsConfirmed": "No",
+                "dateModified": "2017-01-02 20:12:14"
+            },
+            {
+                "idDrink": "11022",
+                "strDrink": "Allies Cocktail",
+                "strDrinkAlternate": null,
+                "strTags": null,
+                "strVideo": null,
+                "strCategory": "Ordinary Drink",
+                "strIBA": null,
+                "strAlcoholic": "Alcoholic",
+                "strGlass": "Cocktail glass",
+                "strInstructions": "Stir all ingredients with ice, strain contents into a cocktail glass, and serve.",
+                "strInstructionsES": null,
+                "strInstructionsDE": "Alle Zutaten mit Eis verrühren, Inhalt in ein Cocktailglas abseihen und servieren.",
+                "strInstructionsFR": null,
+                "strInstructionsIT": "Mescolare tutti gli ingredienti con ghiaccio, filtrare il contenuto in una coppetta da cocktail e servire.",
+                "strInstructionsZH-HANS": null,
+                "strInstructionsZH-HANT": null,
+                "strDrinkThumb": "https://www.thecocktaildb.com/images/media/drink/qvprvp1483388104.jpg",
+                "strIngredient1": "Dry Vermouth",
+                "strIngredient2": "Gin",
+                "strIngredient3": "Kummel",
+                "strIngredient4": null,
+                "strIngredient5": null,
+                "strIngredient6": null,
+                "strIngredient7": null,
+                "strIngredient8": null,
+                "strIngredient9": null,
+                "strIngredient10": null,
+                "strIngredient11": null,
+                "strIngredient12": null,
+                "strIngredient13": null,
+                "strIngredient14": null,
+                "strIngredient15": null,
+                "strMeasure1": "1 oz ",
+                "strMeasure2": "1 oz ",
+                "strMeasure3": "1/2 tsp ",
+                "strMeasure4": null,
+                "strMeasure5": null,
+                "strMeasure6": null,
+                "strMeasure7": null,
+                "strMeasure8": null,
+                "strMeasure9": null,
+                "strMeasure10": null,
+                "strMeasure11": null,
+                "strMeasure12": null,
+                "strMeasure13": null,
+                "strMeasure14": null,
+                "strMeasure15": null,
+                "strImageSource": null,
+                "strImageAttribution": null,
+                "strCreativeCommonsConfirmed": "No",
+                "dateModified": "2017-01-02 20:15:04"
+            }
+        ]
+    }
+    const [searchText, setSearchText] = useState<string>("")
+
+    const filteredCocktails = cocktailsResponse.drinks.filter(cocktail => cocktail.strDrink.toLocaleLowerCase().includes(searchText.toLocaleLowerCase()))
+
+    const onSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setSearchText(event.target.value)
```

Comment:
Ich glaube ich würde das in eine Search-Componente auslagern, dann ist es hier etwas übersichtlicher. Aber ist nur ein Vorschlag